var/spack/repos/builtin/packages/pumi/package.py

```diff
@@ -0,0 +1,64 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import sys
+
+
+class Pumi(CMakePackage):
+    """SCOREC RPI's Parallel Unstructured Mesh Infrastructure (PUMI).
+       An efficient distributed mesh data structure and methods to support
+       parallel adaptive analysis including general mesh-based operations,
+       such as mesh entity creation/deletion, adjacency and geometric
+       classification, iterators, arbitrary (field) data attachable to mesh
+       entities, efficient communication involving entities duplicated
+       across multiple tasks, migration of mesh entities between tasks,
+       and dynamic load balancing."""
+
+    homepage = "https://www.scorec.rpi.edu/pumi"
+    url      = "https://github.com/SCOREC/core.git"
+
+    version('master', git='https://github.com/SCOREC/core.git',
```

Comment:
What @davydden means is that Spack uses version numbers for things like sorting. If I ask for `pumi@2.5:`, Spack needs to know whether the version is greater than or less than 2.5. Right now, `develop` is considered to be _higher_ than any other numbered version, while all other non-numeric strings are considered to be _less_ than any numbered versions. I personally view this as a bug and think `master ` and `develop` should both be treated as greater than all numeric versions, but that started a long argument that spans multiple issues and fell to the wayside as there were bigger fish to fry. For what it's worth, I personally add `master` versions to Spack and don't care much for version sorting if there's only one version.